src/messaging/serializers.py

```diff
@@ -19,11 +19,32 @@ class ThreadSerializer(WritableNestedModelSerializer):
 	profile_1 = ProfileSerializer()
 	profile_2 = ProfileSerializer()
 	recent_message = serializers.SerializerMethodField()
-
+	
 	def get_recent_message(self, obj):
 		return MessageSerializer(Message.objects.filter(thread__id=obj.id).order_by('timestamp').last()).data
 
 	class Meta:
 		model = Thread
 		fields = ('id', 'profile_1', 'profile_2', 'recent_message', )
 		read_only_fields = ('id', 'profile_1', 'profile_2', 'recent_message', )
+
+
+class ThreadListSerializer(WritableNestedModelSerializer):
```

Comment:
This isn't serializing a list.  Let's just call it `OwnThreadSerializer` and slap a comment on the top saying you need to supply context of the request.